types/react-router-redux/index.d.ts

```diff
@@ -31,7 +31,7 @@ export interface RouterState {
     location: Location | null;
 }
 
-export function routerReducer(state?: RouterState, action?: RouterAction): RouterState;
+export function routerReducer(state?: RouterState, action?: RouterAction | AnyAction): RouterState;
```

Comment:
Actually `routerReducer` as any other reducer would accept any possible action that Redux accepts. It is not true that `action` passed to `routerReducer` would be `RouterAction`. All actions apart from router's will be ignored, but they are still a valid input for the reducer.

I'd suggest using `Reducer` type:

```ts
export const routerReducer: Reducer<RouterState>;
```